REST service

SignatureBeef opened the pull request:
Adds an Owin REST service
Configurations now save defaults when created
Adds the Service attribute to PlanTextAccountService

SignatureBeef committed:
Owin rest service

Needs reviewing, particularly for:
IRestService - is this required since plugins will need to reference
owin anyway
.csproj and package changes

SignatureBeef committed:
Inject the configuration service instead of requesting

SignatureBeef committed:
Removed unneeded packages from launcher

SignatureBeef committed:
Service atrribute for PlainTextAccountService

SignatureBeef committed:
Save() on new configurations

SignatureBeef committed:
Merge remote-tracking branch 'refs/remotes/NyxStudios/general-devel' into general-devel

Ijwu commented on Orion/Rest/Owin/OwinRestService.cs:
```diff
@@ -0,0 +1,116 @@
+		/// Occurs when OWIN requests the service to configure the HTTP service.
+		/// </summary>
+		/// <param name="appBuilder">The <see cref="IAppBuilder"/> instance for use with configuring the HTTP service</param>
+		private void Startup(IAppBuilder appBuilder)
```
Isn't the use of IAppBuilder deprecated? I can't find any meaningful documentation about the interface itself.

SignatureBeef replied:
@Ijwu You might be right as the docs I was looking at were back from 2013.
It seems ASP .NET Core is where they are heading to (?) which uses IApplicationBuilder. 
If we're on the same page and we don't mind using the mvc packages for the web api + a billion references, I can easily switch this over tonight.
See https://docs.asp.net/en/latest/intro.html

SignatureBeef committed:
Switched from owin to asp net core

SignatureBeef committed:
[travis] try adding the universe repo for libuv-dev

SignatureBeef committed:
Revert "[travis] try adding the universe repo for libuv-dev"

This reverts commit 8f4f91ca017f9e9806f6c88a49a92ee9783e3912.

SignatureBeef committed:
Revert "Switched from owin to asp net core"

This reverts commit 8390e40669bb919619a6dc9965805ce5b8c0ee22.

QuiCM commented:
Tagging in @NyxStudios/tshock for further commentary on this. Are we all good for a conflict resolution + merge?

SignatureBeef committed:
Merge remote-tracking branch 'refs/remotes/NyxStudios/general-devel' into general-devel

# Conflicts:
#	Orion.Launcher/packages.config
#	Orion.Tests/Orion.Tests.csproj
#	Orion.Tests/packages.config
#	Orion/Orion.csproj
#	Orion/packages.config

SignatureBeef committed:
Ninject dependency resolver for the OwinRestService

Implements a custom IDependencyResolver that uses Ninject so
ApiControllers can use dependency injection

QuiCM commented:
@DeathCradle does this now fulfil all of the requests mentioned above?

SignatureBeef commented:
@WhiteXZ The stuff mentioned from @tylerjwatson, yeah it is.
The .NET Core stuff ended up causing too many problems so after a discussion with @tylerjwatson we decided that reverting back to the owin service was best.
